File: src/main/java/com/aladdinworks44/service/PowerSupplyAssignmentService.java
package com.aladdinworks44.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworks44.domain.PowerSupplyAssignment;
import com.aladdinworks44.dto.PowerSupplyAssignmentDTO;
import com.aladdinworks44.dto.PowerSupplyAssignmentSearchDTO;
import com.aladdinworks44.dto.PowerSupplyAssignmentPageDTO;
import com.aladdinworks44.dto.PowerSupplyAssignmentConvertCriteriaDTO;
import com.aladdinworks44.service.GenericService;
import com.aladdinworks44.dto.common.RequestDTO;
import com.aladdinworks44.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PowerSupplyAssignmentService extends GenericService<PowerSupplyAssignment, Integer> {

	List<PowerSupplyAssignment> findAll();

	ResultDTO addPowerSupplyAssignment(PowerSupplyAssignmentDTO powerSupplyAssignmentDTO, RequestDTO requestDTO);

	ResultDTO updatePowerSupplyAssignment(PowerSupplyAssignmentDTO powerSupplyAssignmentDTO, RequestDTO requestDTO);

    Page<PowerSupplyAssignment> getAllPowerSupplyAssignments(Pageable pageable);

    Page<PowerSupplyAssignment> getAllPowerSupplyAssignments(Specification<PowerSupplyAssignment> spec, Pageable pageable);

	ResponseEntity<PowerSupplyAssignmentPageDTO> getPowerSupplyAssignments(PowerSupplyAssignmentSearchDTO powerSupplyAssignmentSearchDTO);
	
	List<PowerSupplyAssignmentDTO> convertPowerSupplyAssignmentsToPowerSupplyAssignmentDTOs(List<PowerSupplyAssignment> powerSupplyAssignments, PowerSupplyAssignmentConvertCriteriaDTO convertCriteria);

	PowerSupplyAssignmentDTO getPowerSupplyAssignmentDTOById(Integer powerSupplyAssignmentId);







}

File: aladdinworks44_ui/src/services/PowerSupplyAssignmentService.js
import http from "../http-common"; 

class PowerSupplyAssignmentService {
  getAllPowerSupplyAssignments(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/powerSupplyAssignment/powerSupplyAssignments`, searchDTO);
  }

  get(powerSupplyAssignmentId) {
    return this.getRequest(`/powerSupplyAssignment/${powerSupplyAssignmentId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/powerSupplyAssignment?field=${matchData}`, null);
  }

  addPowerSupplyAssignment(data) {
    return http.post("/powerSupplyAssignment/addPowerSupplyAssignment", data);
  }

  update(data) {
  	return http.post("/powerSupplyAssignment/updatePowerSupplyAssignment", data);
  }
  
  uploadImage(data,powerSupplyAssignmentId) {
  	return http.postForm("/powerSupplyAssignment/uploadImage/"+powerSupplyAssignmentId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PowerSupplyAssignmentService();
